pkgs/development/libraries/science/math/mkl/default.nix

```diff
@@ -1,74 +1,18 @@
-{ stdenvNoCC, writeText, fetchurl, rpmextract, undmg }:
+{ stdenvNoCC, writeText, fetchurl, rpmextract, undmg, openmp }:
 /*
-  Some (but not all) mkl functions require openmp, but Intel does not add these
-  to SO_NEEDED and instructs users to put openmp on their LD_LIBRARY_PATH. If
-  you are using mkl and your library/application is using some of the functions
-  that require openmp, add a setupHook like this to your package:
-
-  setupHook = writeText "setup-hook.sh" ''
-    addOpenmp() {
-        addToSearchPath LD_LIBRARY_PATH ${openmp}/lib
-    }
-    addEnvHooks "$targetOffset" addOpenmp
-  '';
-
-  We do not add the setup hook here, because avoiding it allows this large
-  package to be a fixed-output derivation with better cache efficiency.
- */
-
-stdenvNoCC.mkDerivation rec {
-  name = "mkl-${version}";
+  Some mkl functions require openmp, but Intel does not add these to SO_NEEDED
+  and instructs users to put openmp on their LD_LIBRARY_PATH. Since the mkl
+  package is large and changes very infrequently, we make a fixed-output pkg
+  with the extracted RPM contents, then wrap it with an openmp-dependent
+  derivation with the appropriate setupHook. Note that we cannot add an
+  SO_NEEDED entry directly to the library, since the license prevents
+  modification.
+*/
+let
   version = "${date}.${rel}";
   date = "2019.0";
   rel = "117";
```

Comment:
If you look at this in split view with whitespace ignored, I didn't really change much; it's just that there's a `let ... in ...` in this file now, so the indentation is different.